### PR TITLE
Implement support for version opening and closing

### PR DESCRIPTION
**Pull request by mjgiarlo**
Refs sul-dlss/argo#1277

### CODE REVIEW

File: lib/dor/services/client.rb [hunk 1]
```diff
@@ -127,6 +127,25 @@ def create_release_tag(object:, release:, what:, to:, who:)
           release_tags.create(object: object, release: release, what: what, to: to, who: who)
         end
 
+        # Open new version for an object
+        # @param object [String] object identifier
+        # @param params [Hash] optional params (see dor-services-app)
+        # @raise [UnexpectedResponse] when the response is not successful.
+        # @raise [MalformedResponse] when the response is not parseable.
+        # @return [String] the current version
+        def open_new_version(object:, **params)
```
Comment (jcoyne) [inline lib/dor/services/client.rb]: I wonder what you think of not having this method and just having the consumer do `client.object(druid).new_version(params`?

File: lib/dor/services/client/objects.rb [hunk 2]
```diff
@@ -42,11 +42,12 @@ def notify_goobi(object:)
         def current_version(object:)
           SDR.new(connection: connection, version: api_version).current_version(object: object)
         end
-        deprecation_deprecate current_version: 'Use Dor::Client.sdr.current_version instead'
+        deprecation_deprecate current_version: 'Use Dor::Services::Client.sdr.current_version(object: obj) instead'
```
Comment (jcoyne) [inline lib/dor/services/client/objects.rb]: Thanks for fixing these.

File: lib/dor/services/client.rb (hunk 1 above)
Comment (mjgiarlo) [reply lib/dor/services/client.rb]: I like the idea of hiding internal client complexity from consumers, and making all methods callable from the main Client class. (At some point, this list of methods may grow cumbersome, sure.)

Comment (mjgiarlo) [reply lib/dor/services/client.rb]: @jcoyne 	If you're cool with this, mind having another look?

**Pull request merged**